bottlerocket/agents/src/sonobuoy.rs

```diff
@@ -5,18 +5,30 @@ use model::{Outcome, TestResults};
 use serde_json::Value;
 use snafu::{ensure, OptionExt, ResultExt};
 use std::collections::HashMap;
+use std::convert::TryInto;
 use std::path::Path;
 use std::process::Command;
 use std::time::Duration;
+use test_agent::InfoClient;
 
 /// Runs the sonobuoy conformance tests according to the provided configuration and returns a test
 /// result at the end.
-pub async fn run_sonobuoy(
+pub async fn run_sonobuoy<I>(
     kubeconfig_path: &str,
     e2e_repo_config_path: Option<&str>,
     sonobuoy_config: &SonobuoyConfig,
     results_dir: &Path,
-) -> Result<TestResults, error::Error> {
+    info_client: &I,
+) -> Result<TestResults, error::Error>
+where
+    I: InfoClient,
+{
+    let mut results = TestResults {
+        outcome: Outcome::InProgress,
+        other_info: Some("Starting test".to_string()),
+        ..Default::default()
+    };
+    info_client.send_test_update(results.clone()).await?;
```

Comment:
Do we want to continue or bomb when errors like this one occur?

```suggestion
    info_client.send_test_update(results.clone()).await.ok_or_else(|e| {
        error!("Unable to sent test update: {}", e);
    });
```